hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/AllowanceValidator.java

```diff
@@ -138,8 +138,7 @@ public static Account getEffectiveOwner(
             @Nullable final AccountID owner,
             @NonNull final Account payer,
             @NonNull final ReadableAccountStore accountStore) {
-        final var ownerNum = owner != null ? owner.accountNumOrElse(0L) : 0L;
-        if (ownerNum == 0 || ownerNum == payer.accountNumber()) {
+        if (owner == null || owner.accountNumOrElse(0L) == 0L || owner == payer.accountId()) {
```

Comment:
Same as before to use equals for AccountID

Reply:
Fixed